src/log.py

```diff
@@ -0,0 +1,63 @@
+"""
+log
+
+By: Norma Chang
+"""
+import json
+import os.path
+import time
+
+
+class Log:
+    """
+    Logging class.
+
+    Logs data from vehicle.
+    """
+    def __init__(self, vehicle_name):
+        self.file_name = ""
+        self.class_name = ""
+        self._start_time = time.strftime('%b %d %Y %H:%M:%S', time.gmtime())
+        self._start_time = self._start_time.replace(" ", "").replace(":", "")
+        if vehicle_name.lower() == 'follower':
+            self.class_name = vehicle_name
+        elif vehicle_name.lower() == 'leader':
+            self.class_name = vehicle_name
+        else:
+            raise ValueError('Unexpected class name')
+
+        self.file_name = self.class_name + self._start_time + '.log'
+
+
+    def write_to_file(self, dictionary):
+        """
+        Write vehicle data into text file.
+
+        :param: dictionary: vehicle dictionary
+        :type: dictionary
+        """
+        self.vehicle_dict = {
+                'Time': time.time(),
+                'Dictionary': dictionary
+            } 
+        with open(self.file_name, 'a') as outfile:
+            json.dump(self.vehicle_dict, outfile)
+            outfile.write(',')
+
+
+    def read_log_file(self, file_name):
+        """
+        Reads the log file.
+
+        :param: file_name: Unique, auto-generated file name.
```

Comment:
This doesn't have to be a unique, auto-generated name. It needs to be a valid name.  We should be able to read any log file that is formatted in the way we want it to be formatted. You should mention that.

Reply:
Tried to include this in the docstring